### ISSUE TITLE
About /config/target.yaml

### ISSUE THREAD

**Issue by AllenYkl** [gjhhust/YOLOFT#8]
Hi there, I tried to reproduce the training phase, I took a look at the initial data cfg - "config/target.yaml", I don't really understand the train, test split claimed in it, and I also browsed in your provided XS-VID dataset, I still did not find the corresponding file. Could you help me plz ? Thx a lot! 
![Screenshot 2024-10-16 at 13 51 39](https://github.com/user-attachments/assets/667b5df0-8b60-498b-97a0-3f42b57cbce7)

**Comment by gjhhust** [comment]
Regarding data configuration, you can refer to the README.md file at https://github.com/gjhhust/YOLOFT/blob/main/README.md

![Screenshot_20241016_140524_com huawei browser_edit_317273835852106](https://github.com/user-attachments/assets/49334d5a-9119-4cfa-bf4c-f2b9d9b02412).

The configuration file specifically for XSVID is located at: https://github.com/gjhhust/YOLOFT/blob/main/config/XS-VID.yaml

It's worth noting that annotation files are preferably in both YOLO and COCO formats (though not strictly necessary). The YOLO format is used for training, while the COCO format is provided for testing (eval_json) to enable the possibility of outputting COCO evaluation metrics.

If you want to use your own dataset, you can follow these steps:

Generate basic annotations in COCO format, where the `file_name` field in the `images` section should use relative addresses such as "videos_name/framenumber.jpg or png", for example, "video1/0000123.png". This is because during subsequent training, the absolute address will be constructed as `images_dir/file_name` for image retrieval.
Convert the COCO annotations to YOLO format using the `coco2yolo.py` file provided within the XSVID dataset. This will generate the `labels_dir` and the YOLO annotation files within it.
Fill in the `data.yaml` file to proceed with training.

The final data organization format should resemble the one shown in the above figure.

**Comment by AllenYkl** [comment]
Thank you so much for your help!

**Comment by gjhhust** [comment]
you can refer to the coco.json file in XSVID for the COCO format JSON get more detail.

**Comment by AllenYkl** [comment]
Sry another question, for the test.txt and train.txt, do you have any scripts to generate them? 

> you can refer to the coco.json file in XSVID for the COCO format JSON get more detail.

**Comment by AllenYkl** [comment]
> Sry another question, for the test.txt and train.txt, do you have any scripts to generate them?
> 
> > you can refer to the coco.json file in XSVID for the COCO format JSON get more detail.

Or I guess I can generate them from the "file_name" attribute in coco.json?

**Comment by gjhhust** [comment]
no problem, you can also use coco2yolo.py, this scripts can generate 'xxx.txt'(if your json named xxx.json), and generate yolo format in 'lables_dir'

**Comment by gjhhust** [comment]
Yes, you're right. The train.txt and test.txt files contain the addresses of the images, and these files are used to index the images during training. The labels_dir contains the annotation files (.txt) for each frame of the images, which follow the YOLO format as specified by the Ultralytics official usage.

**Comment by gjhhust** [comment]
> Thx a lot for your patience!

No problem.
